internal/core/graphics/image.rs

```diff
@@ -780,4 +788,44 @@ pub(crate) mod ffi {
     ) {
         core::ptr::write(image, Image::from(ImageInner::StaticTextures(textures)));
     }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn slint_image_compare_equal(image1: &Image, image2: &Image) -> bool {
+        return image1.eq(image2);
+    }
+}
+
+/// This structure contains fields to identify and render an OpenGL texture that Slint borrows from the applicatin code.
+/// Use this to embed a native OpenGL texture into a Slint scene, by creating the texture with [`Self::new()`] and
+/// converting it into an [`Image`] using `.into()`.
+///
+/// The ownership of the texture remains with the application. It is the application's responsibility to delete the texture
+/// when it is not used anymore.
+///
+/// Note that only 2D RGBA textures are supported.
+#[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
+#[cfg(not(target_arch = "wasm32"))]
+#[repr(C)]
+pub struct BorrowedOpenGLTexture {
```

Comment:
FWIW, I don't like this to be a top-level type, so my preference would be to move this into a `gpu` module (so `slint::gpu::BorrowedOpenGLTexture`.

Reply:
Why do we even need this type? Couldn't we just have  Image::from_borrowed_opengl_texture(texture_id, size) ?

Reply:
Sure, but there will be a more in the future though - EGL external image, Vulkan, YUV. But if you don't mind then I'll add them all to `slint::Image`.

Reply:
I think  several `Image::from_xyz`, `Image::from_hjk`, ... is actually more discoverable than many classes in a gpu modules. Don't you think?

Reply:
I don't feel too strongly about it, so let's start with `from_xyz` - we can always revise this later.